lsp-mode.el

```diff
@@ -8385,6 +8385,43 @@ This avoids overloading the server with many files when starting Emacs."
     (lsp--error "Nothing to disconnect from?")))
 
 
+;;; `lsp-iedit'
+(lsp-defun lsp--range-text ((&RangeToPoint :start :end))
+  (buffer-substring start end))
+
+(defun lsp-iedit--on-ranges (ranges)
+  "Start an `iedit' operation using RANGES.
+RANGES shall be a list of lsp-`&Range's. They can be acquired
+from various lsp protocol requests, e.g.
+`textDocument/documentHighlight', ...."
+  (unless (seq-empty-p ranges)
+    (mapc (-lambda ((&RangeToPoint :start :end))
+            (push (iedit-make-occurrence-overlay start end)
+                  iedit-occurrences-overlays))
+          ranges)
+    ;; See `iedit-start'; TODO: upstream this
+    (setq iedit-mode t)
+    (when iedit-auto-buffering
+      (iedit-start-buffering))
+    (run-hooks 'iedit-mode-hook)
+    (add-hook 'before-revert-hook 'iedit-done nil t)
+    (add-hook 'kbd-macro-termination-hook 'iedit-done nil t)
+    (add-hook 'change-major-mode-hook 'iedit-done nil t)
+    (add-hook 'iedit-aborting-hook 'iedit-done nil t)
+    (message "%d occurrences of \"%s\""
+             (seq-length ranges)
+             (lsp--range-text (lsp-seq-first ranges)))))
+
+(defun lsp-iedit-highlights ()
+  "Start an `iedit' operation on the documentHighlights at point.
+This can be used as a primitive `lsp-rename' replacement if the
+language server doesn't support renaming.
+
+See also `lsp-enable-symbol-highlighting'."
+  (interactive)
+  (let ((highlights (lsp-request "textDocument/documentHighlight"
```

Comment:
We should check if user has `iedit` and if is required via `(require 'iedit nil t)`
Also we should declare the `iedit` methods, check how we do for `avy` integration

Reply:
I will do that.

Reply:
This will probably fix the CI

Reply:
Fixed. Although `(require 'iedit nil t)` is not correct here, as that function would fail much less intuitively if `iedit` wasn't available.